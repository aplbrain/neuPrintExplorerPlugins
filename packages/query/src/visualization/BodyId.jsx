import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Icon from '@material-ui/core/Icon';
import Modal from '@material-ui/core/Modal';
import Tooltip from '@material-ui/core/Tooltip';
import { SunburstLoader } from '@neuprint/support';
import SelectAndCopyText from '../shared/SelectAndCopyText';

const styles = theme => ({
  icon: {
    marginLeft: '3px',
    marginTop: '3px',
    cursor: 'pointer',
    color: theme.palette.primary.main
  },
  nblink: {
    marginLeft: '3px',
    textDecoration: 'none',
    fontWeight: 'bold',
    color: theme.palette.primary.main
  },
  container: {
    display: 'flex',
    flexDirection: 'row'
  },
  paper: {
    position: 'absolute',
    top: '50%',
    left: '50%',
    transform: 'translate(-50%, -50%)',
    width: '80%',
    height: '85%',
    display: 'block',
    backgroundColor: theme.palette.background.paper,
    boxShadow: theme.shadows[5],
    padding: theme.spacing(4),
    outline: 'none'
  }
});

/**
 * Launches actions for opening the skeleton viewer and neuroglancer.
 *
 * @param {string} id
 * @param {string} dataset
 * @param {Object} actions
 */
function showSkeleton(id, dataset, actions, color) {
  actions.neuroglancerAddandOpen(id, dataset);
  actions.skeletonAddandOpen(id, dataset, null, color);
}

function BodyId(props) {
  const { children, dataSet, actions, classes, options={skeleton: true} } = props;
  const [modal, setModal] = useState(false);
  const neuronbridgeLink = `https://neuronbridge.janelia.org/search?q=${dataSet.replace(/:.*$/, '*')}:${children}`;
  return (
    <div>
      <div className={classes.container}>
        <SelectAndCopyText text={children} actions={actions} />
        {options.skeleton ? (
        <Tooltip title="Skeleton View">
          <Icon
            className={classes.icon}
            onClick={() => showSkeleton(children, dataSet, actions, options.color)}
            fontSize="inherit"
          >
            visibility
          </Icon>
        </Tooltip>) : ""}
        <Tooltip title="Synapse Connectivity">
          <Icon className={classes.icon} onClick={() => setModal(!modal)} fontSize="inherit">
            donut_small
          </Icon>
        </Tooltip>
        {/hemibrain|manc/.test(dataSet) && (
          <Tooltip title="NeuronBridge">
            <a className={classes.nblink} href={neuronbridgeLink}>
              NB
            </a>
          </Tooltip>
        )}
      </div>
      <Modal open={modal} onClose={() => setModal(false)}>
        <div className={classes.paper}>
          <SunburstLoader bodyId={children} dataSet={dataSet} />
        </div>
      </Modal>
    </div>
  );
}

BodyId.propTypes = {
  children: PropTypes.number.isRequired,
  dataSet: PropTypes.string.isRequired,
  actions: PropTypes.object.isRequired,
  classes: PropTypes.object.isRequired,
  options: PropTypes.object
};

BodyId.defaultProps = {
  options: {
    color: null,
    skeleton: true
  },
};

export default withStyles(styles)(BodyId);
