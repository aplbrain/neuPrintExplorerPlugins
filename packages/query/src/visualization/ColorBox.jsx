import React from 'react';
import PropTypes from 'prop-types';

function ColorBox({ margin, width, height, backgroundColor, title, text, color='#000' }) {
  const styles = {
    margin: `${margin}px`,
    width: `${width}px`,
    minWidth: `${width}px`,
    height: `${height}px`,
    minHeight: `${height}px`,
    backgroundColor,
    color,
    overflow: 'visible',
    display: 'inline-flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  };
  return (
    <div style={styles} title={title}>
      {text}
    </div>
  );
}

ColorBox.propTypes = {
  margin: PropTypes.number.isRequired,
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  text: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired
};

export default ColorBox;
