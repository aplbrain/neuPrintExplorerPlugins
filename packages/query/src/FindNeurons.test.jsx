import { FindNeurons } from './FindNeurons';
import { ColorLegend } from './visualization/MiniRoiHeatMap';

let wrapper;
let button;
let textField;
let inputSelect;
let outputSelect;
let limitNeuronsToggle;
let preThresholdField;
let postThresholdField;

const styles = { select: {}, clickable: {} };
const { actions, React, enzyme, renderer } = global;

const neoServerSettings = {
  get: () => 'http://example.com'
};

const component = (
  <FindNeurons
    availableROIs={['roiA', 'roiB', 'roiC']}
    dataSet="test"
    datasetstr="test"
    actions={actions}
    classes={styles}
    history={{ push: jest.fn() }}
    isQuerying={false}
    neoServerSettings={neoServerSettings}
    urlQueryString=""
    neoServer="testServer"
  />
);

describe('find neurons Plugin', () => {
  beforeAll(() => {
    wrapper = enzyme.mount(component);
    button = wrapper.find('Button');
    textField = wrapper.find('TextField').at(0);
    limitNeuronsToggle = wrapper.find('Switch');
    preThresholdField = wrapper.find('TextField').at(1);
    postThresholdField = wrapper.find('TextField').at(2);
    inputSelect = wrapper.find('Select').at(0);
    outputSelect = wrapper.find('Select').at(1);
  });
  beforeEach(() => {
    actions.submit.mockClear();
  });
  it('has name and description', () => {
    expect(FindNeurons.queryName).toBeTruthy();
    expect(FindNeurons.queryDescription).toBeTruthy();
  });
  it('renders correctly', () => {
    const pluginView = renderer.create(component).toJSON();
    expect(pluginView).toMatchSnapshot();
  });
  describe('when user clicks submit', () => {
    it('should return a query object and submit', () => {
      expect(button.props().onClick()).toEqual(
        expect.objectContaining({
          dataSet: 'test',
          queryString: '/npexplorer/findneurons',
          visType: 'SimpleTable',
          visProps: { rowsPerPage: 25 },
          plugin: 'FindNeurons',
          parameters: {
            dataset: 'test',
            input_ROIs: [],
            output_ROIs: [],
            all_segments: false,
            statuses: []
          },
          title: 'Neurons with inputs in [] and outputs in []',
          menuColor: expect.any(String),
          processResults: expect.any(Function)
        })
      );
      expect(actions.submit).toHaveBeenCalledTimes(1);

      // if neuron name/id is present add to parameters
      textField.props().onChange({ target: { value: 'abc' } });
      expect(button.props().onClick()).toEqual(
        expect.objectContaining({
          dataSet: 'test',
          menuColor: expect.any(String),
          parameters: {
            dataset: 'test',
            input_ROIs: [],
            neuron_name: 'abc',
            output_ROIs: [],
            all_segments: false,
            statuses: []
          },
          plugin: 'FindNeurons',
          processResults: expect.any(Function),
          queryString: '/npexplorer/findneurons',
          title: 'Neurons with inputs in [] and outputs in []',
          visProps: { rowsPerPage: 25 },
          visType: 'SimpleTable'
        })
      );
      textField.props().onChange({ target: { value: '123' } });
      expect(button.props().onClick()).toEqual(
        expect.objectContaining({
          dataSet: 'test',
          menuColor: expect.any(String),
          parameters: {
            dataset: 'test',
            input_ROIs: [],
            neuron_id: 123,
            output_ROIs: [],
            all_segments: false,
            statuses: []
          },
          plugin: 'FindNeurons',
          processResults: expect.any(Function),
          queryString: '/npexplorer/findneurons',
          title: 'Neurons with inputs in [] and outputs in []',
          visProps: { rowsPerPage: 25 },
          visType: 'SimpleTable'
        })
      );

      // if input/output rois present add to parameters
      textField.props().onChange({ target: { value: '' } });
      inputSelect.props().onChange([{ value: 'roiA' }]);
      outputSelect.props().onChange([{ value: 'roiB' }]);
      expect(button.props().onClick()).toEqual(
        expect.objectContaining({
          dataSet: 'test',
          menuColor: expect.any(String),
          parameters: {
            dataset: 'test',
            input_ROIs: ['roiA'],
            output_ROIs: ['roiB'],
            all_segments: false,
            statuses: []
          },
          plugin: 'FindNeurons',
          processResults: expect.any(Function),
          queryString: '/npexplorer/findneurons',
          title: 'Neurons with inputs in [roiA] and outputs in [roiB]',
          visProps: { rowsPerPage: 25 },
          visType: 'SimpleTable'
        })
      );

      // if neuron/segment filters present add to parameters
      limitNeuronsToggle.props().onChange();
      preThresholdField.props().onChange({ target: { value: 12 } });
      postThresholdField.props().onChange({ target: { value: 13 } });
      expect(button.props().onClick()).toEqual(
        expect.objectContaining({
          dataSet: 'test',
          menuColor: expect.any(String),
          parameters: {
            dataset: 'test',
            input_ROIs: ['roiA'],
            output_ROIs: ['roiB'],
            all_segments: true,
            pre_threshold: 12,
            post_threshold: 13,
            statuses: []
          },
          plugin: 'FindNeurons',
          processResults: expect.any(Function),
          queryString: '/npexplorer/findneurons',
          title: 'Neurons with inputs in [roiA] and outputs in [roiB]',
          visProps: { rowsPerPage: 25 },
          visType: 'SimpleTable'
        })
      );
    });
    it('should process returned results into data object', () => {
      const query = {
        dataSet: 'test',
        queryString: '/npexplorer/findneurons',
        visType: 'SimpleTable',
        plugin: 'FindNeurons',
        visProps: { rowsPerPage: 25 },
        parameters: {
          dataset: 'test',
          input_ROIs: [],
          output_ROIs: [],
          all_segments: false,
          statuses: []
        },
        title: 'Neurons with inputs in [] and outputs in []'
      };
      const apiResponse = {
        data: [
          [
            1,
            'KC-s',
            'Traced',
            '{"roiA":{"pre":22,"post":28},"roiB":{"pre":23,"post":31},"roiC":{"pre":45,"post":61}}',
            37325787,
            90,
            120,
            ['roiA', 'roiB', 'roiC'],
            true
          ]
        ],
        columns: [],
        debug: 'test'
      };
      const processedResults = wrapper.instance().processResults(query, apiResponse);
      expect(processedResults).toEqual(
        expect.objectContaining({
          columns: [
            'id',
            'neuron',
            'status',
            '#post (inputs)',
            '#pre (outputs)',
            '#voxels',
            <div>
              roi heatmap <ColorLegend />
            </div>,
            'roi breakdown'
          ],
          data: expect.arrayContaining([]),
          debug: 'test'
        })
      );

      // if no data returned
      const processedResultsEmpty = wrapper.instance().processResults(query, {
        columns: [],
        data: [],
        debug: 'test'
      });
      expect(processedResultsEmpty).toEqual({
        columns: [
          'id',
          'neuron',
          'status',
          '#post (inputs)',
          '#pre (outputs)',
          '#voxels',
          <div>
            roi heatmap <ColorLegend />
          </div>,
          'roi breakdown'
        ],
        data: [],
        debug: 'test'
      });

      // if rois selected should add roi count columns
      const queryWithRoisSelected = {
        dataSet: 'test',
        queryString: '/npexplorer/findneurons',
        visType: 'SimpleTable',
        plugin: 'FindNeurons',
        visProps: { rowsPerPage: 25 },
        parameters: {
          dataset: 'test',
          input_ROIs: ['roiA'],
          output_ROIs: ['roiB'],
          all_segments: false,
          statuses: []
        },
        title: 'Neurons with inputs in [] and outputs in []'
      };
      const processedResultsWithRois = wrapper
        .instance()
        .processResults(queryWithRoisSelected, apiResponse);
      expect(processedResultsWithRois).toEqual(
        expect.objectContaining({
          columns: [
            'id',
            'neuron',
            'status',
            '#post (inputs)',
            '#pre (outputs)',
            'roiA #post',
            'roiA #pre',
            'roiB #post',
            'roiB #pre',
            '#voxels',
            <div>
              roi heatmap <ColorLegend />
            </div>,
            'roi breakdown'
          ],
          data: expect.arrayContaining([]),
          debug: 'test'
        })
      );
    });
  });
  describe('when user hits enter key', () => {
    it('should submit request', () => {
      const processRequest = jest.spyOn(wrapper.instance(), 'processRequest');
      const preventDefault = jest.fn();
      textField.props().onKeyDown({ keyCode: 13, preventDefault });
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(processRequest).toHaveBeenCalledTimes(1);
      expect(actions.submit).toHaveBeenCalledTimes(1);
    });
  });
  describe('when user inputs text or selects rois', () => {
    it('should change url query string in state', () => {
      actions.setQueryString.mockClear();

      // neuron name input
      textField.props().onChange({ target: { value: 'abc' } });
      expect(actions.setQueryString).toHaveBeenCalledTimes(1);
      expect(actions.getQueryObject('fn').neuronName).toBe('abc');

      // input rois
      inputSelect.props().onChange([{ value: 'roiA' }, { value: 'roiB' }]);
      expect(actions.getQueryObject('fn').inputROIs).toContainEqual('roiA');
      expect(actions.getQueryObject('fn').inputROIs).toContainEqual('roiB');
      expect(actions.getQueryObject('fn').inputROIs.length).toBe(2);
      expect(actions.setQueryString).toHaveBeenCalledTimes(2);

      // output rois
      outputSelect.props().onChange([{ value: 'roiB' }, { value: 'roiC' }]);
      expect(actions.getQueryObject('fn').outputROIs).toContainEqual('roiC');
      expect(actions.getQueryObject('fn').outputROIs).toContainEqual('roiB');
      expect(actions.getQueryObject('fn').outputROIs.length).toBe(2);
      expect(actions.setQueryString).toHaveBeenCalledTimes(3);
    });
  });
});
