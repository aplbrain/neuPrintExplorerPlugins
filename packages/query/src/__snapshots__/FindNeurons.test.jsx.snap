// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`find neurons Plugin renders correctly 1`] = `
<div>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="select-multiple-chip"
  >
    Input ROIs
  </label>
  <div
    className="css-10nd86i [object Object]"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-5-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="select-multiple-chip"
  >
    Output ROIs
  </label>
  <div
    className="css-10nd86i [object Object]"
    onKeyDown={[Function]}
  >
    <div
      className="css-vj8t7z"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="css-1hwfws3"
      >
        <div
          className="css-1492t68"
        >
          Select...
        </div>
        <div
          className="css-1g6gooi"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="react-select-6-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="css-1wy0on6"
      >
        <span
          className="css-d8oujb"
        />
        <div
          aria-hidden="true"
          className="css-1ep9fjw"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-19bqh2r"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiFormControl-root-19 MuiFormControl-fullWidth-22"
  >
    <div>
      <div
        className="MuiFormControl-root-19 MuiFormControl-fullWidth-22"
        onKeyDown={[Function]}
      >
        <label
          className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-formControl-6 MuiInputLabel-animated-9"
          data-shrink={false}
        >
          Neuron name or ID (optional)
        </label>
        <div
          className="MuiInputBase-root-37 MuiInput-root-24 MuiInput-underline-28 MuiInputBase-fullWidth-46 MuiInput-fullWidth-31 MuiInputBase-formControl-38 MuiInput-formControl-25 MuiInputBase-multiline-45 MuiInput-multiline-30"
          onClick={[Function]}
        >
          <div
            className="MuiPrivateTextarea-root-54"
          >
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-55 MuiPrivateTextarea-shadow-56"
              readOnly={true}
              rows="1"
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-hidden="true"
              className="MuiPrivateTextarea-textarea-55 MuiPrivateTextarea-shadow-56"
              readOnly={true}
              rows={1}
              tabIndex={-1}
              value=""
            />
            <textarea
              aria-invalid={false}
              className="MuiPrivateTextarea-textarea-55 MuiInputBase-input-47 MuiInput-input-32 MuiInputBase-inputMultiline-49 MuiInput-inputMultiline-34"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              rows={1}
              style={
                Object {
                  "height": 19,
                }
              }
              value=""
            />
          </div>
        </div>
      </div>
      <p
        aria-describedby={null}
        className="MuiTypography-root-65 MuiTypography-body1-74 MuiTypography-colorError-99 NeuronHelp-badge-23"
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        title="Enter body ID, neuron name, or wildcard names using period+star (e.g., MBON.*). Warning: if using regular expressions, special characters like parentheses must be escaped (e.g. Delta6g\\\\\\\\(preQ7\\\\\\\\).*)"
      >
        ?
      </p>
    </div>
  </div>
  <div
    className="NeuronFilter-expandablePanel-101"
  >
    <div
      className="MuiPaper-root-109 MuiPaper-elevation1-112 MuiExpansionPanel-root-106"
    >
      <div
        aria-expanded={false}
        className="MuiButtonBase-root-142 MuiExpansionPanelSummary-root-136"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex="0"
      >
        <div
          className="MuiExpansionPanelSummary-content-140"
        >
          <h6
            className="MuiTypography-root-65 MuiTypography-subtitle1-83"
          >
            Optional neuron/segment filters
          </h6>
        </div>
        <div
          aria-hidden="true"
          className="MuiButtonBase-root-142 MuiIconButton-root-145 MuiExpansionPanelSummary-expandIcon-141"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label-150"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-151"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-225"
          />
        </div>
      </div>
      <div
        aria-hidden="true"
        className="MuiCollapse-container-160"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper-162"
        >
          <div
            className="MuiCollapse-wrapperInner-163"
          >
            <div
              className="MuiExpansionPanelDetails-root-164 NeuronFilter-nopad-102"
            >
              <div
                className="MuiFormControl-root-19 NeuronFilter-formControl-100"
              >
                <div
                  className="MuiFormControl-root-19 NeuronFilter-formControl-100"
                >
                  <label
                    className="MuiFormControlLabel-root-165"
                  >
                    <span
                      className="MuiSwitch-root-171"
                    >
                      <span
                        className="MuiButtonBase-root-142 MuiIconButton-root-145 MuiPrivateSwitchBase-root-180 MuiSwitch-switchBase-174 MuiSwitch-colorPrimary-176 MuiPrivateSwitchBase-checked-181 MuiSwitch-checked-175"
                        onBlur={[Function]}
                        onContextMenu={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        onKeyUp={[Function]}
                        onMouseDown={[Function]}
                        onMouseLeave={[Function]}
                        onMouseUp={[Function]}
                        onTouchEnd={[Function]}
                        onTouchMove={[Function]}
                        onTouchStart={[Function]}
                        tabIndex={null}
                      >
                        <span
                          className="MuiIconButton-label-150"
                        >
                          <span
                            className="MuiSwitch-icon-172 MuiSwitch-iconChecked-173"
                          />
                          <input
                            checked={true}
                            className="MuiPrivateSwitchBase-input-183"
                            disabled={false}
                            onChange={[Function]}
                            type="checkbox"
                          />
                        </span>
                        <span
                          className="MuiTouchRipple-root-225"
                        />
                      </span>
                      <span
                        className="MuiSwitch-bar-179"
                      />
                    </span>
                    <span
                      className="MuiTypography-root-65 MuiTypography-body1-74 MuiFormControlLabel-label-170"
                    >
                      <h6
                        className="MuiTypography-root-65 MuiTypography-subtitle1-83"
                        style={
                          Object {
                            "display": "inline-flex",
                          }
                        }
                      >
                        Limit to neurons
                        <div
                          aria-describedby={null}
                          className="NeuronFilter-tooltip-105"
                          onBlur={[Function]}
                          onFocus={[Function]}
                          onMouseLeave={[Function]}
                          onMouseOver={[Function]}
                          onTouchEnd={[Function]}
                          onTouchStart={[Function]}
                          title="Limit to neurons (bodies with >=2 t-bars, >=10 psds, name, soma, or status)"
                        >
                          ?
                        </div>
                      </h6>
                    </span>
                  </label>
                </div>
                <div
                  className="MuiFormControl-root-19 MuiFormControl-marginDense-21 NeuronFilter-textField-104"
                >
                  <label
                    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-marginDense-7 MuiInputLabel-outlined-11"
                    data-shrink={false}
                  >
                    minimum # pre (optional)
                  </label>
                  <div
                    className="MuiInputBase-root-37 MuiOutlinedInput-root-184 MuiInputBase-formControl-38 MuiInputBase-marginDense-44"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="MuiPrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline-191"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="MuiPrivateNotchedOutline-legend-198"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-47 MuiOutlinedInput-input-192 MuiInputBase-inputType-50 MuiInputBase-inputMarginDense-48 MuiOutlinedInput-inputMarginDense-193"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      rows={1}
                      type="number"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root-19 MuiFormControl-marginDense-21 NeuronFilter-textField-104"
                >
                  <label
                    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-formControl-6 MuiInputLabel-animated-9 MuiInputLabel-marginDense-7 MuiInputLabel-outlined-11"
                    data-shrink={false}
                  >
                    minimum # post (optional)
                  </label>
                  <div
                    className="MuiInputBase-root-37 MuiOutlinedInput-root-184 MuiInputBase-formControl-38 MuiInputBase-marginDense-44"
                    onClick={[Function]}
                  >
                    <fieldset
                      aria-hidden={true}
                      className="MuiPrivateNotchedOutline-root-197 MuiOutlinedInput-notchedOutline-191"
                      style={
                        Object {
                          "paddingLeft": 8,
                        }
                      }
                    >
                      <legend
                        className="MuiPrivateNotchedOutline-legend-198"
                        style={
                          Object {
                            "width": 0.01,
                          }
                        }
                      >
                        <span
                          dangerouslySetInnerHTML={
                            Object {
                              "__html": "&#8203;",
                            }
                          }
                        />
                      </legend>
                    </fieldset>
                    <input
                      aria-invalid={false}
                      className="MuiInputBase-input-47 MuiOutlinedInput-input-192 MuiInputBase-inputType-50 MuiInputBase-inputMarginDense-48 MuiOutlinedInput-inputMarginDense-193"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      required={false}
                      rows={1}
                      type="number"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="MuiFormControl-root-19 NeuronFilter-formControl-100"
                >
                  <label
                    className="MuiFormLabel-root-12"
                    style={
                      Object {
                        "display": "inline-flex",
                      }
                    }
                  >
                    Filter by status
                    <div
                      aria-describedby={null}
                      className="NeuronFilter-tooltip-105"
                      onBlur={[Function]}
                      onFocus={[Function]}
                      onMouseLeave={[Function]}
                      onMouseOver={[Function]}
                      onTouchEnd={[Function]}
                      onTouchStart={[Function]}
                      title=""
                    >
                      ?
                    </div>
                  </label>
                  <div
                    className="css-10nd86i NeuronFilter-select-103"
                    onKeyDown={[Function]}
                  >
                    <div
                      className="css-vj8t7z"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <div
                        className="css-1hwfws3"
                      >
                        <div
                          className="css-1492t68"
                        >
                          Select...
                        </div>
                        <div
                          className="css-1g6gooi"
                        >
                          <div
                            className=""
                            style={
                              Object {
                                "display": "inline-block",
                              }
                            }
                          >
                            <input
                              aria-autocomplete="list"
                              autoCapitalize="none"
                              autoComplete="off"
                              autoCorrect="off"
                              disabled={false}
                              id="react-select-7-input"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              spellCheck="false"
                              style={
                                Object {
                                  "background": 0,
                                  "border": 0,
                                  "boxSizing": "content-box",
                                  "color": "inherit",
                                  "fontSize": "inherit",
                                  "opacity": 1,
                                  "outline": 0,
                                  "padding": 0,
                                  "width": "1px",
                                }
                              }
                              tabIndex="0"
                              type="text"
                              value=""
                            />
                            <div
                              style={
                                Object {
                                  "height": 0,
                                  "left": 0,
                                  "overflow": "scroll",
                                  "position": "absolute",
                                  "top": 0,
                                  "visibility": "hidden",
                                  "whiteSpace": "pre",
                                }
                              }
                            >
                              
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        className="css-1wy0on6"
                      >
                        <span
                          className="css-d8oujb"
                        />
                        <div
                          aria-hidden="true"
                          className="css-1ep9fjw"
                          onMouseDown={[Function]}
                          onTouchEnd={[Function]}
                        >
                          <svg
                            aria-hidden="true"
                            className="css-19bqh2r"
                            focusable="false"
                            height={20}
                            viewBox="0 0 20 20"
                            width={20}
                          >
                            <path
                              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root-142 MuiButton-root-199 MuiButton-contained-210 MuiButton-containedPrimary-211 MuiButton-raised-213 MuiButton-raisedPrimary-214"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-200"
    >
      Submit
    </span>
    <span
      className="MuiTouchRipple-root-225"
    />
  </button>
</div>
`;
